orc8r/cloud/go/services/certifier/servicers/certifier.go

```diff
@@ -768,6 +801,33 @@ func checkAction(reqResource *certprotos.Resource, policyResource *certprotos.Re
 	return certprotos.Effect_UNKNOWN
 }
 
+// checkResource checks if the requested resource is authorized by the policy
+func checkResource(reqResource, policyResource *certprotos.Resource) certprotos.Effect {
+	// The policy does not handle this case, so return UNKNOWN.
+	if reqResource.GetResourceId() != nil &&
+		policyResource.ResourceType != certprotos.ResourceType_URI &&
+		reqResource.ResourceType != policyResource.ResourceType {
+		return certprotos.Effect_UNKNOWN
+	}
+	finalEffect := certprotos.Effect_UNKNOWN
+	if ok, _ := doublestar.Match(policyResource.Resource, reqResource.Resource); ok {
+		finalEffect = policyResource.Effect
+	}
+
+	resourceID := ""
+	if reqResource.ResourceType == certprotos.ResourceType_NETWORK_ID {
+		resourceID = reqResource.GetNetworkId()
+	}
+	if reqResource.ResourceType == certprotos.ResourceType_TENANT_ID {
+		resourceID = reqResource.GetTenantId()
+	}
+	if finalEffect != certprotos.Effect_DENY && resourceID == policyResource.Resource {
+		finalEffect = policyResource.Effect
+	}
```

Comment:
Hmmm should I create two separate types for reqResource and policyResource?
Right now what's happening is:
reqResource.resource is always a path & reqResource.resourceID is the network/tenantID
policyResource.resource could be path, networkID, and tenantID